llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -1324,7 +1322,12 @@ void DwarfDebug::finalizeModuleInfo() {
     // ranges for all subprogram DIEs for mach-o.
     DwarfCompileUnit &U = SkCU ? *SkCU : TheCU;
 
-    if (unsigned NumRanges = TheCU.getRanges().size()) {
+    if (Asm->TM.getTargetTriple().isNVPTX() && tuneForGDB()) {
+      // PTX does not support subtracting labels from the code section in the
+      // debug_loc section.  To work around this, the NVPTX backend needs the
+      // compile unit to have no low_pc in order to have a zero base_address
+      // when handling debug_loc in cuda-gdb.
```

Comment:
I've moved the PTX test inside the other block, which is probably easier to read.  I don't want to merge the two conditions, exactly, since one is an assignment, and a type-preserving merge would be ugly there.